streets_reader/src/split_ways.rs

```diff
@@ -121,37 +88,29 @@ pub fn split_up_roads(
                     }
                 }
 
+                let i1_node_id = input.osm_node_ids.get(&pts[0].to_hashable()).cloned();
+                let i2_node_id = input.osm_node_ids.get(&pt.to_hashable()).cloned();
+
                 let untrimmed_center_line = simplify_linestring(std::mem::take(&mut pts));
                 match PolyLine::new(untrimmed_center_line) {
                     Ok(pl) => {
+                        // TODO If either endpoint is on the boundary, we won't record any OSM ID.
```

Comment:
I want to look into this as a followup. OK with getting rid of `OriginalRoad` entirely? A caller can look at a Road's 0 or more WayIDs and for each endpoint, 0 or more NodeIDs. There's lots of possibilities there to link to original OSM thing, and callers need to deal with it